text/3857-cfg-version.md

```diff
@@ -0,0 +1,634 @@
+you would instead use `"1.27.0-0"` to match all pre-release versions of 1.27.0:
+```rust
+#[cfg_attr(since(rust, "1.27.0-0"), must_use)]
+fn double(x: i32) -> i32 {
+    2 * x
+}
+
+fn main() {
+    double(4);
+    // warning: unused return value of `double` which must be used
+    // ^--- This warning only happens if we are on Rust >= 1.27.
+}
+```
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+## `since` cfg predicate
+
+A `since` cfg predicate will be added to Rust.
+As Cargo mirrors Rust's `#[cfg]` syntax, it too will gain this predicate.
+
+The [syntax](https://doc.rust-lang.org/reference/conditional-compilation.html#grammar-ConfigurationPredicate) is:
+```
+ConfigurationVersion -> `since` `(` IDENTIFIER `,` ( STRING_LITERAL | RAW_STRING_LITERAL ) `)`
+```
+
+If `IDENTIFIER` has no value or is undefined, this will evaluate to `false`.
+
+If `IDENTIFIER` is not a valid [SemVer](https://semver.org/) value, this will evaluate to `false`.
+
+If the string literal does not conform to the syntax from `<major>` to `<major>.<minor>.<patch>-<pre-release>` where the first three fields must be integers, this will evaluate to `false`.
+Note that this excludes support for the `+build` field.
+
+Otherwise, the `IDENTIFIER` will be compared to the string literal according to
+[Cargo's `>=` version requirements](https://doc.rust-lang.org/nightly/cargo/reference/specifying-dependencies.html#comparison-requirements).
+For example, `#[cfg(since(rust, "1.90"))]` will be treated as `1.95.2 >= 1.90.0`.
+
+See also `--check-cfg`.
+
+## `--check-cfg`
+
+A new predicate will be added of the form:
+```
+ConfigurationVersion -> `since` `(` ( STRING_LITERAL | RAW_STRING_LITERAL ) `)`
+```
+
+The syntax for the contents of the string literal is a SemVer value without the build field.
+
+This will specify that the given `--cfg` is valid for all values that match:
+- SemVer syntax
+- from the specified version and up
+
+When the given `--cfg` is used with the `since` predicate:
+- the string literal should not be of a syntax that evaluates to `false`
+- the minimum version requirement must specify a subset of what the `--check-cfg` specifies
+
+So given `--check-cfg 'cfg(foo, values(since("1.95.0")))'`,
+- ✅ `#[cfg(foo = "1.100.0")]`
+- ⚠️ `#[cfg(foo = "1.0")]`: not SemVer syntax
+- ✅ `#[cfg(since(foo, "1.95.0"))]`
+- ✅ `#[cfg(since(foo, "1.100.0"))]`
+- ✅ `#[cfg(since(foo, "3.0.0"))]`
+- ✅ `#[cfg(since(foo, "1.95"))]`
+- ⚠️ `#[cfg(since(foo, "1.90.0"))]`: matches a superset of `--check-cfg`
+- ⚠️ `#[cfg(since(foo, "1.95.0-0"))]`: matches a superset of `--check-cfg`
+- ⚠️ `#[cfg(since(foo, "1"))]`: matches a superset of `--check-cfg`
+- ⚠️ `#[cfg(since(foo, "bar"))]`: invalid string literal syntax
+
+## `rust` cfg
+
+A new built-in cfg `--cfg rust=<version>` will be added by the compiler
+that specifies the language version.
```

Comment:
Would the `<version>` here be quoted? because if is, then that would be a breaking change as we would start interpreting the inside of the quoted string, potentially breaking user who might do `--cfg my_cfg="126.0"` and expect `#[cfg(my_cfg = "1.127")]` to not match.

Reply:
This does not change anything about how cfg's are specified.